### PR TITLE
kind: 0.2.1 -> v0.3.0

### PR DESCRIPTION
**Pull request by j0xaf**
###### Motivation for this change

Upgrade kind from 0.2.1 to v0.3.0.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] other Linux distributions
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by j0xaf** [comment]
cc @gilligan and @thoughtpolice who may be interested in this

**Comment by gilligan** [comment]
@GrahamcOfBorg build kind

File: pkgs/development/tools/kind/default.nix [hunk 1]
```diff
@@ -4,18 +4,19 @@ with stdenv.lib;
 
 buildGoPackage rec {
   name = "kind-${version}";
-  version = "0.2.1";
+  version = "v0.3.0";
```
Comment (nlewo) [inline pkgs/development/tools/kind/default.nix]: According to the manual, the version should start by a digit.
```suggestion
  version = "0.3.0";
```

File: pkgs/development/tools/kind/default.nix [hunk 2]
```diff
@@ -4,18 +4,19 @@ with stdenv.lib;
...
 
   src = fetchFromGitHub {
     rev    = "${version}";
```
Comment (nlewo) [inline pkgs/development/tools/kind/default.nix]:
```suggestion
    rev    = "v${version}";
```

**Review by nlewo — CHANGES_REQUESTED** [review]

**Comment by thoughtpolice** [comment]
LGTM.

**Pull request merged**